source_pubsub/client.go

```diff
@@ -255,39 +259,56 @@ func (c *PubSubClient) Wait(ctx context.Context) error {
 	}
 }
 
+func (c *PubSubClient) ResetRetryCount() {
+	c.retryCount = c.maxRetries
+}
+
 func (c *PubSubClient) ReplayID() []byte {
 	return c.currReplayID
 }
 
+func (c *PubSubClient) retryAuth(ctx context.Context, retry bool) (error, bool) {
```

Comment:
it may be easier to use this library to retry https://github.com/avast/retry-go

Reply:
Added the retry library